tests/MongoDB.Driver.Examples/InsertDataWithEncryptedFieldsWithAutoEncryption.cs

```diff
@@ -0,0 +1,295 @@
+                keyVaultNamespace: keyVaultNamespace,
+                kmsProviders: kmsProviders,
+                schemaMap: schemaMap,
+                extraOptions: extraOptions);
+            mongoClientSettings.AutoEncryptionOptions = autoEncryptionSettings;
+            return new MongoClient(mongoClientSettings);
+        }
+
+        [Fact]
+        public void GenerateJsonSchema()
+        {
+            // 4. Code snippet: generate the JSON Schema and assign it to a variable
+
+            var schema = JsonSchemaCreator.CreateJsonSchema("<your_key_id>").ToString(); // replace "your_key_id" with your base64 data encryption key id 
+        }
+
+        // https://gist.github.com/DmitryLukyanov/8417d1f5c30f12ffeb2be29056aff455#file-autoencryptionconfiguration-cs
+        [Fact]
+        public void AutoEncryptionConfiguration()
+        {
+            AutoEncryptionOptions autoEncryptionOptions = null;
+
+            var extraOptions = new Dictionary<string, object>()
+            {
+                { "mongocryptdSpawnArgs", new [] { "--port=30000" } },
+            };
+            autoEncryptionOptions.With(extraOptions: extraOptions);
+        }
+
+        // https://gist.github.com/DmitryLukyanov/8417d1f5c30f12ffeb2be29056aff455#file-insertpatient-cs
+#pragma warning disable xUnit1013 // Public method should be marked as test
+        public static void InsertPatient(
+#pragma warning restore xUnit1013 // Public method should be marked as test
+            IMongoCollection<BsonDocument> collection,
+            string name,
+            int ssn,
+            string bloodType,
+            BsonDocument[] medicalRecords,
+            int policyNumber,
+            string provider)
+        {
+            var insurance = new BsonDocument
+            {
+                { "policyNumber", policyNumber },
+                { "provider", provider }
+            };
+
+            var patient = new BsonDocument
+            {
+                { "name", name },
+                { "ssn", ssn },
+                { "bloodType", bloodType },
+                { "medicalRecords", BsonArray.Create(medicalRecords) },
+                { "insurance", insurance }
+            };
+
+            collection.InsertOne(patient);
+        }
+    }
+
+    // https://gist.github.com/DmitryLukyanov/8417d1f5c30f12ffeb2be29056aff455#file-jsonschemacreator-cs
+    public static class JsonSchemaCreator
+    {
+        private static readonly string DETERMINISTIC_ENCRYPTION_TYPE = "AEAD_AES_256_CBC_HMAC_SHA_512-Deterministic";
+        private static readonly string RANDOM_ENCRYPTION_TYPE = "AEAD_AES_256_CBC_HMAC_SHA_512-Random";
+
+        private static BsonDocument CreateEncryptMetadata(string base64KeyId)
+        {
+            var guid = GuidConverter.FromBytes(Convert.FromBase64String(base64KeyId), GuidRepresentation.Standard);
+            var binary = new BsonBinaryData(guid, GuidRepresentation.Standard);
```

Comment:
Replace lines 242-243 with:

            var keyId = new BsonBinaryData(Convert.FromBase64String(keyIdBase64), BsonBinarySubType.UuidStandard);
            return new BsonDocument("keyId", new BsonArray(new[] { keyId }));

Reply:
If the above replacement doesn't work (error messages during serialization) then some configuration setting is probably wrong.

Reply:
I would investigate it in the scope of this ticket.

The code as written (creating a Guid first) should not be necessary and if you are finding that it is necessary that is not something that should be glossed over. It is a problem that should be fixed.

Remember this is documentation. This is where we are telling users HOW they should do something. Let's get it right.

Reply:
Done, but to make it work we need to merge changes from `CSHARP-3252`